### DISCUSSION TITLE
Preventing use of `false` as an array key

### DISCUSSION THREAD

**Discussion by M1ke** [vimeo/psalm#5965] [Ideas]
https://psalm.dev/r/b5c9862354

In the given example we are wanting to search for an instance of a string in a list and unset it. However due to PHP's type coercion we accidentally unset the first element in our list instead because `array_search` returns `false`, and unsetting `false` coerces to `0`.

As far as I see Psalm isn't flagging anything here.

I believe Psalm should be able to flag if a potential boolean value is used as an array key as some form of error - bool false is never an appropriate array-key - if somebody wishes to use potential `false` as an array key they should type cast it (e.g. `'false'` would be a legitimate array key, as would `0`)

**Comment by orklah** [comment]
Seems like a case where the `ignoreInternalFunctionFalseReturn` config should make a difference, but it doesn't
